CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ else()
   )
 endif()
 
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
Does it work well with construct at line below?

```cmake
  if(CMAKE_MINOR_VERSION VERSION_GREATER "3.18")
    # Ask for 20, may get anything below
    set(CMAKE_CXX_STANDARD 20)
  else()
    # Ask for 17, may get anything below
    set(CMAKE_CXX_STANDARD 17)
  endif()
```

i.e. if I'm asking for 20, but my compiler only supports 17. Would I get 17 without any error?